lib/sauce-service.js

```diff
@@ -110,12 +110,30 @@ class SauceService {
     /**
      * update Sauce Labs job
      */
-    after () {
+    after (result) {
         if (!this.sauceUser || !this.sauceKey) {
             return
         }
+        let failures = this.failures
 
-        return this.updateJob(this.sessionId, this.failures)
+        /**
+         * set failures if user has bail option set in which case afterTest and
+         * afterSuite aren't executed before after hook
+         */
+        if (global.browser.config.mochaOpts && global.browser.config.mochaOpts.bail && Boolean(result)) {
+            failures = 1
+        }
+
+        // const status = 'status: ' + (failures > 0 ? 'failing' : 'passing')
```

Comment:
Not sure what do about the logging methods as that was not previously done here in this service it looks like?